qiskit_experiments/curve_analysis/data_processing.py

```diff
@@ -116,6 +119,20 @@ def mean_xy_data(
 
         return x_means, y_means, y_sigmas
 
+    # Quadrature sum of variance
+    if method == "quad_sum":
+        for i in range(x_means.size):
+            # Get positions of y to average
+            idxs = xdata == x_means[i]
+            ys = ydata[idxs]
+            ss = sigma[idxs]
+
+            # Compute sample mean and biased sample variance
+            y_means[i] = np.mean(ys)
+            y_sigmas[i] = np.sqrt(np.mean(ss ** 2))
```

Comment:
Discussing, we have concluded that if the number of shots in each of the averaged experiments is identical, we can calculate the variance as follows:
```
n_points = len(idxs)
y_sigmas[i] = np.sqrt((1 / n_points) ** 2 * np.sum(ss ** 2))
```
Or:
``y_sigmas[i] = (1 / n_points) * np.sqrt(np.sum(ss ** 2))``

Reply:
Possibly, if it is possible to extract the number of shots used for each experiment, we can write the more accurate formula as something like:
```
# Assume that exp_shots holds the number of shots for each experiment
n_points = len(idxs)
total_shots = np.sum(exp_shots)
weights = exp_shots / total_shots
y_sigmas[i] = np.sqrt(np.sum((weights ** 2) * (ss ** 2)))
```

Reply:
Thanks @haggaila . Done in https://github.com/Qiskit/qiskit-experiments/pull/409/commits/41afbde5b3eddfd51b796d23e11c769909003946. Could you please double check if the calculation is correct?

Reply:
@nkanazawa1989 yes, the implementation looks definitely good.